Userland/Applications/PDFViewer/PDFViewerWidget.cpp

```diff
@@ -37,21 +42,119 @@ void PDFViewerWidget::initialize_menubar(GUI::Menubar& menubar)
     file_menu.add_action(GUI::CommonActions::make_quit_action([](auto&) {
         GUI::Application::the()->quit();
     }));
+}
 
-    auto& view_menu = menubar.add_menu("&View");
+void PDFViewerWidget::create_toolbar()
+{
+    auto& toolbar_container = add<GUI::ToolbarContainer>();
+    auto& toolbar = toolbar_container.add<GUI::Toolbar>();
 
-    auto open_sidebar_action = GUI::Action::create(
+    auto open_outline_action = GUI::Action::create(
         "Open &Sidebar", { Mod_Ctrl, Key_O }, Gfx::Bitmap::load_from_file("/res/icons/16x16/sidebar.png"), [&](auto& action) {
             m_sidebar_open = !m_sidebar_open;
             m_sidebar->set_fixed_width(m_sidebar_open ? 0 : 200);
             action.set_text(m_sidebar_open ? "Open &Sidebar" : "Close &Sidebar");
         },
         nullptr);
-    open_sidebar_action->set_enabled(false);
+    open_outline_action->set_enabled(false);
+    m_toggle_sidebar_action = open_outline_action;
+
+    toolbar.add_action(*open_outline_action);
+    toolbar.add_separator();
+
+    m_go_to_prev_page_action = GUI::Action::create("Go to &Previous Page", Gfx::Bitmap::load_from_file("/res/icons/16x16/go-up.png"), [&](auto&) {
+        if (m_viewer->current_page() <= 0)
+            return;
+
+        auto new_page_index = m_viewer->current_page() - 1;
+        m_viewer->set_current_page(new_page_index);
+        m_viewer->update();
+        m_page_text_box->set_text(String::number(new_page_index + 1));
+    });
+    m_go_to_prev_page_action->set_enabled(false);
+
+    m_go_to_next_page_action = GUI::Action::create("Go to &Next Page", Gfx::Bitmap::load_from_file("/res/icons/16x16/go-down.png"), [&](auto&) {
+        if (m_viewer->current_page() >= m_viewer->document()->get_page_count() - 1)
+            return;
+
+        auto new_page_index = m_viewer->current_page() + 1;
+        m_viewer->set_current_page(new_page_index);
+        m_viewer->update();
+        m_page_text_box->set_text(String::number(new_page_index + 1));
+    });
+    m_go_to_next_page_action->set_enabled(false);
+
+    toolbar.add_action(*m_go_to_prev_page_action);
+    toolbar.add_action(*m_go_to_next_page_action);
+
+    m_page_text_box = toolbar.add<GUI::TextBox>();
+    m_page_text_box->set_enabled(false);
+    m_page_text_box->set_fixed_width(30);
 
-    view_menu.add_action(open_sidebar_action);
+    m_page_text_box->on_up_pressed = [&] {
+        auto current_index_opt = m_page_text_box->text().to_int();
+        if (!current_index_opt.has_value()) {
+            m_page_text_box->set_text("1");
```

Comment:
That seems a bit surprising, maybe just filter out non-numeric characters and set the text to that?
Should make it appear as though you can't write non-numeric stuff into that text bar